pkg/agent/controller/noderoute/node_route_controller.go

```diff
@@ -352,6 +370,16 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 	for i := 0; i < defaultWorkers; i++ {
 		go wait.Until(c.worker, time.Second, stopCh)
 	}
+
+	go func() {
+		// When the initial list of Nodes has been processed, we decrement flowRestoreCompleteWait.
+		defer c.flowRestoreCompleteWait.Done()
+		wait.PollUntilContextCancel(wait.ContextForChannel(stopCh), 100*time.Millisecond, true, func(ctx context.Context) (done bool, err error) {
```

Comment:
Although it will unlikely happen, it's possible that stopCh receives signal before `HasSynced` return true, then it shouldn't log ""Initial list of Nodes has been processed"" and decrement the wait group?

Reply:
I think it's a valid point. I'll update the code and maybe add a unit test for this case.